packages/components/src/utils/unit-values.ts

```diff
@@ -0,0 +1,48 @@
+/**
+ * Parses a number and unit from a value.
+ *
+ * @param  toParse Value to parse
+ *
+ * @return  The extracted number and unit.
+ */
+export function parseUnitValue(
+	toParse: string | number
+): [ number | string | undefined, string | undefined ] {
+	// eslint-disable-next-line eqeqeq
+	if ( toParse == null ) {
+		return [ undefined, undefined ];
+	}
+
+	const value = String( toParse ).trim();
+
+	let num: number | string = parseFloat( value );
+	num = Number.isNaN( num ) ? '' : num;
+
+	const matched = value.match(
+		/[\d.\-+]*\s*(cm|mm|Q|in|pc|pt|px|em|ex|ch|rem|lh|vw|vh|vmin|vmax)/
```

Comment:
Should we specify in comments / JSDoc that by "Unit" we intend a valid CSS unit?

Reply:
Also, TIL about `Q` (Quarter-millimeters) !